src/net.cpp

```diff
@@ -2786,38 +2803,29 @@ CNode::~CNode()
     CloseSocket(hSocket);
 }
 
-void CNode::AskFor(const CInv& inv)
+void CNode::AddToAskSet(const CInv& inv)
 {
-    if (mapAskFor.size() > MAPASKFOR_MAX_SZ || setAskFor.size() > SETASKFOR_MAX_SZ)
+    if (ask_set.size() > ASKSET_MAX_SZ)
         return;
     // a peer may not have multiple non-responded queue positions for a single inv item
-    if (!setAskFor.insert(inv.hash).second)
+    if (!ask_set.insert(inv).second)
         return;
 
-    // We're using mapAskFor as a priority queue,
-    // the key is the earliest time the request can be sent
-    int64_t nRequestTime;
-    limitedmap<uint256, int64_t>::const_iterator it = mapAlreadyAskedFor.find(inv.hash);
-    if (it != mapAlreadyAskedFor.end())
-        nRequestTime = it->second;
-    else
-        nRequestTime = 0;
-    LogPrint(BCLog::NET, "askfor %s  %d (%s) peer=%d\n", inv.ToString(), nRequestTime, FormatISO8601Time(nRequestTime/1000000), id);
-
     // Make sure not to reuse time indexes to keep things in the same order
     int64_t nNow = GetTimeMicros() - 1000000;
     static int64_t nLastTime;
     ++nLastTime;
     nNow = std::max(nNow, nLastTime);
     nLastTime = nNow;
 
-    // Each retry is 2 minutes after the last
-    nRequestTime = std::max(nRequestTime + 2 * 60 * 1000000, nNow);
-    if (it != mapAlreadyAskedFor.end())
-        mapAlreadyAskedFor.update(it, nRequestTime);
-    else
-        mapAlreadyAskedFor.insert(std::make_pair(inv.hash, nRequestTime));
-    mapAskFor.insert(std::make_pair(nRequestTime, inv));
+    tx_to_ask.insert(std::make_pair(inv.hash, nNow));
+
+    int priority = rand() % OUTBOUND_ASK_PRIORITY;
+    if (!fInbound) {
+        priority += OUTBOUND_ASK_PRIORITY;
```

Comment:
I am a bit confused: 
So here if a node is inbound - it's priority is bound by OUTBOUND_ASK_PRIORITY,
and if it is outbound - you add OUTBOUND_ASK_PRIORITY one more time.

So outbound nodes will have higher priority value, right? 

But then in 
`std::multimap<int, std::pair<CNode*, CInv>>::iterator peerToAsk = peersToAskTx.begin();`
You take the node with lowest priority?

Reply:
oops. You are right.